### ISSUE TITLE
Safe navigator operator (&.) breaks ABC check?

### ISSUE THREAD

**Issue by rachel-carvalho** [bbatsov/rubocop#3578]
I just noticed that when I added a safe navigator operator (`&.`) `rubocop` stopped reporting a method's ABC size. If anything, there should be even more branches, not less.

I updated to the latest version of rubocop and still see the problem.

Fortunately, I was able to reproduce it with bogus code.

With the safe navigator I have no warning, but without it, I get `C: Assignment Branch Condition size for why? is too high. [15.07/15]`, as expected. Snippets are available on the steps to reproduce section.

--------

## Expected behavior

It should report `C: Assignment Branch Condition size for why? is too high. [15.07/15]`

## Actual behavior

It finds no issues with the method after I introduce a `&.`.

## Steps to reproduce the problem

- 1. Run rubocop on this method:
```ruby
def why?
  @instance1 = other_method
  @instance2 = other_method

  @instance3 = {}

  return if @instance1.blank? || @instance2.blank?

  @instance4 = other_method
  @instance5 = @defined_somewhere_else.method.method

  @instance6 = @instance4.method ? @instance4.other_method : other_method

  @instance7 = @instance6.method { |i| i.method.method }
end
```
- 2. See how it reports `C: Assignment Branch Condition size for why? is too high. [15.07/15]`
- 3. Now add a safe navigator operator on the last line:
```ruby
def why?
  @instance1 = other_method
  @instance2 = other_method

  @instance3 = {}

  return if @instance1.blank? || @instance2.blank?

  @instance4 = other_method
  @instance5 = @defined_somewhere_else.method.method

  @instance6 = @instance4.method ? @instance4.other_method : other_method

  @instance7 = @instance6&.method { |i| i.method.method }
end
```
- 4. See how it no longer reports a problem.

## RuboCop version
```
0.43.0 (using Parser 2.3.1.4, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by savef** [comment]
I think this line is the issue:
https://github.com/bbatsov/rubocop/blob/ec3123f/lib/rubocop/cop/metrics/abc_size.rb#L14

We need to add `:csend` as an alternate branch node, as per [the docs](https://github.com/whitequark/parser/blob/master/doc/AST_FORMAT.md).

**Comment by rachel-carvalho** [comment]
Thanks so much for such a quick fix 🎉 @savef @bbatsov
